pkg/server/router/router.go

```diff
@@ -188,3 +197,36 @@ func (router *router) registerPipelineRecordAPIs(ws *restful.WebService) {
 		Param(ws.PathParameter("pipeline", "name of the pipeline").DataType("string")).
 		Param(ws.PathParameter("recordId", "id of the pipeline record").DataType("string")))
 }
+
+// registerScmAPIs registers scm related endpoints.
+func (router *router) registerScmAPIs(ws *restful.WebService) {
+	logdog.Info("Register scm APIs")
+
+	ws.Path(APIVersion).Consumes(restful.MIME_JSON).Produces(restful.MIME_JSON)
+	// GET /api/v1/projects/{project}/scm/{resource}/token
```

Comment:
What about resource -> type?  It is not a resource, but a scm type.

Reply:
done